components/locid_transform/src/directionality.rs

```diff
@@ -0,0 +1,147 @@
+///     .expect("create failed");
+///
+/// let locale = locale!("en");
+/// assert_eq!(ld.get(locale), Some(Direction::LeftToRight));
+/// ```
+///
+/// [`CLDR`]: http://cldr.unicode.org/
+#[derive(Debug)]
+pub struct LocaleDirectionality {
+    rtl: DataPayload<ScriptDirectionV1Marker>,
+    expander: LocaleExpander,
+}
+
+impl LocaleDirectionality {
+    /// A constructor which takes a [`DataProvider`] and creates a [`LocaleDirectionality`].
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    /// <div class="stab unstable">
+    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
+    /// </div>
+    pub fn try_new_unstable<P>(provider: &P) -> Result<LocaleDirectionality, LocaleTransformError>
+    where
+        P: DataProvider<ScriptDirectionV1Marker>
+            + DataProvider<LikelySubtagsForLanguageV1Marker>
+            + DataProvider<LikelySubtagsForScriptRegionV1Marker>
+            + ?Sized,
+    {
+        let expander = LocaleExpander::try_new_unstable(provider)?;
+        Self::try_new_with_expander_unstable(provider, expander)
+    }
+
+    icu_provider::gen_any_buffer_constructors!(
+        locale: skip,
+        options: skip,
+        error: LocaleTransformError
+    );
+
+    /// Creates a [`LocaleDirectionality`] with a custom [`LocaleExpander`] object.
+    pub fn try_new_with_expander_unstable<P>(
+        provider: &P,
+        expander: LocaleExpander,
+    ) -> Result<LocaleDirectionality, LocaleTransformError>
+    where
+        P: DataProvider<ScriptDirectionV1Marker> + ?Sized,
+    {
+        let rtl: DataPayload<ScriptDirectionV1Marker> =
+            provider.load(Default::default())?.take_payload()?;
+
+        Ok(LocaleDirectionality { rtl, expander })
+    }
+
+    /// Returns the script direction of the given locale.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locid::locale;
+    /// use icu_locid_transform::{Direction, LocaleDirectionality};
+    ///
+    /// let ld = LocaleDirectionality::try_new_unstable(&icu_testdata::unstable())
+    ///     .expect("create failed");
+    ///
+    /// let locale = locale!("en-US");
+    /// assert_eq!(ld.get(locale), Some(Direction::LeftToRight));
+    ///
+    /// let locale = locale!("ar");
+    /// assert_eq!(ld.get(locale), Some(Direction::RightToLeft));
+    ///
+    /// let locale = locale!("fr-Brai-FR");
+    /// assert_eq!(ld.get(locale), None);
+    /// ```
+    pub fn get(&self, locale: impl Into<Locale>) -> Option<Direction> {
+        let mut locale = locale.into();
+
+        let script = locale.id.script.or_else(|| {
+            self.expander.maximize(&mut locale);
+            locale.id.script
+        });
+        let script = match script {
+            Some(script) => script,
+            // If the locale has no script, we cannot determine the directionality.
+            None => return None,
+        };
```

Comment:
```suggestion
        })?;
```

Reply:
D'oh!

Reply:
Surprised clippy didn't catch this, maybe because of the comment.